qiskit_nature/problems/second_quantization/molecular/fermionic_op_builder.py

```diff
@@ -0,0 +1,87 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Fermionic operator builder. """
+
+import itertools
+
+from qiskit_nature.drivers.qmolecule import QMolecule
+from qiskit_nature.operators import FermionicOp
+
+
+def build_fermionic_op(q_molecule: QMolecule) -> FermionicOp:
+    """
+    Builds a fermionic operator based on a QMolecule object.
+
+    Args:
+        q_molecule (QMolecule): QMolecule instance with 1- and/or 2-body integrals.
+
+    Returns:
+        FermionicOp: FermionicOp built from a QMolecule object.
+    """
+
+    one_body_ints = q_molecule.one_body_integrals
+    two_body_ints = q_molecule.two_body_integrals
+
+    return build_ferm_op_from_ints(one_body_ints, two_body_ints)
+
+
+def build_ferm_op_from_ints(one_body_integrals, two_body_integrals=None) -> FermionicOp:
```

Comment:
I assume that you made `two_body_integrals` optional because this was also the case in `FermionicOperator`. I am actually not sure why this is necessary. @woodsp-ibm @pbark do you have any insight on this?

Reply:
Operators for dipole, spin, num particles etc were all just h1 (no h2).

Reply:
Yes, but looking at the code they still always specify `h2 = np.zeros((modes, modes, modes, modes))`.
So I don't think we actually used the fact that `h2` is optional.

Reply:
Currently, dipole case uses h1 only in the code. 
To be honest, I didn't realise that `h2 = np.zeros((modes, modes, modes, modes))` is a dummy value - thought it is somehow relevant for calculations. In such a case, I will delete this variable for num particles and spin so that we benefit fully from having an optional parameter.

Reply:
I do not know why those operators that FermionicOperator could build out specified the same value as the default. I guess in the docstring it doesn't say what h2=None really does, you only get to see by looking at code. Something hopefully we can improve upon with the new code.